src/page/index.ts

```diff
@@ -25,69 +27,95 @@ const cooldown = getEnvVariable('COOLDOWN', envVars) as number;
-
-  recaptcha.verify(req.body['g-recaptcha-response']).then(function (response) {
-    const isHuman = response.isHuman;
+const recaptchaSecretKey = getEnvVariable('GOOGLE_CAPTCHA_PRIVATE', envVars);
+const recaptchaSiteKey = getEnvVariable('GOOGLE_CAPTCHA_SITE_KEY', envVars);
 
-    if (isHuman) {
-      try {
-        decodeAddress(address);
-      } catch (e) {
-        res.status(400).send('Unparsable address.');
-        return;
-      }
+function sendResponse(
+  isHuman: boolean,
+  address: string | Uint8Array | null | undefined,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  response: Response<any, any>
+) {
+  if (isHuman) {
+    try {
+      decodeAddress(address);
+    } catch (e) {
+      response.status(400).send('Unparsable address.');
+      return;
+    }
 
-      const attemptTime = Date.now();
+    const attemptTime = Date.now();
 
-      if (attemptTime - cooldown > lastSuccess) {
-        ax.post<DripResponse>('/page-endpoint', {
-          address,
-          amount: dripAmount,
-        })
-          .then((inner_res) => {
-            if (isDripSuccessResponse(inner_res.data)) {
-              res
-                .status(201)
-                .send(`{"message": "Sent ${dripAmount} ${unit}s."}`);
-              lastSuccess = attemptTime;
-            } else {
-              res
-                .status(429)
-                .send(
-                  inner_res.data.error ||
-                    'An unexpected error occured, please contact us and complain'
-                );
-              return;
-            }
-          })
-          .catch((e) => {
-            res
-              .status(500)
+    if (attemptTime - cooldown > lastSuccess) {
+      ax.post<DripResponse>('/page-endpoint', {
+        address,
+        amount: dripAmount,
+      })
+        .then((inner_res) => {
+          if (isDripSuccessResponse(inner_res.data)) {
+            response
+              .status(201)
+              .send(`{"message": "Sent ${dripAmount} ${unit}s."}`);
+            lastSuccess = attemptTime;
+          } else {
+            response
+              .status(429)
               .send(
-                'An unexpected error occured, please contact us and complain'
+                inner_res.data.error ||
+                  'An unexpected error occurred, please contact us and complain'
               );
-          });
-      } else {
-        res.status(503).send('Too many faucet requests.');
-      }
+            return;
+          }
+        })
+        .catch(() => {
+          response
+            .status(500)
+            .send(
+              'An unexpected error occurred, please contact us and complain'
+            );
+        });
     } else {
-      res.status(401).send('Google captcha failed.');
-      return;
+      response.status(503).send('Too many faucet requests.');
     }
+  } else {
+    response.status(401).send('Google captcha failed.');
+    return;
+  }
```

Comment:
Suggestion: Invert this if, so that this failure is close to the condition that triggers it. Maybe you can even do an early `return` and flatten one level of nesting? Because this `if` is so large, it's difficult to figure out what happens when. Another option would be to pull out some of these internals into functions so that the `if` is smaller.

Reply:
Reworked long function into smaller ones